Xamarin.Forms.Core/RadioButton.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[RenderWith(typeof(_RadioButtonRenderer))]
+	public partial class RadioButton : Button
+	{
+		static Dictionary<string, List<WeakReference<RadioButton>>> _groupNameToElements;
+
+		public const string IsCheckedVisualState = "IsChecked";
+
+		public static readonly BindableProperty IsCheckedProperty = BindableProperty.Create(
+			nameof(IsChecked), typeof(bool), typeof(RadioButton), false, propertyChanged: (b, o, n) => ((RadioButton)b).OnIsCheckedPropertyChanged((bool)n));
+
+		public static readonly BindableProperty GroupNameProperty = BindableProperty.Create(
+			nameof(GroupName), typeof(string), typeof(RadioButton), null, propertyChanged: (b, o, n) => ((RadioButton)b).OnGroupNamePropertyChanged((string)o, (string)n));
+
+		public static readonly BindableProperty ButtonSourceProperty = BindableProperty.Create(
```

Comment:
how does this works ? is this a spriteMap ? don't we need a source for both on and off ?